frontend/packages/dev-console/src/components/project-access/hooks.ts

```diff
@@ -0,0 +1,41 @@
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { ClusterRoleModel } from '@console/internal/models';
+import { K8sResourceCommon } from '@console/internal/module/k8s';
+
+export type ClusterRoleKind = K8sResourceCommon & {
+  rules: { [key: string]: string[] }[];
+};
+
+export enum accessRoles {
+  admin = 'Admin',
+  edit = 'Edit',
+  view = 'View',
+}
+// for testing until backend support
+// eslint-disable-next-line dot-notation
+window.SERVER_FLAGS['availableClusterRoles'] = ['admin', 'view', 'edit', 'cluster-reader'];
+export const useProjectAccessRoles = () => {
+  // eslint-disable-next-line dot-notation
+  const availableClustersRoles = window.SERVER_FLAGS['availableClusterRoles'];
```

Comment:
Small typo, let us call this availableClusterRoles similar to the server flag.

```suggestion
  const availableClusterRoles = window.SERVER_FLAGS['availableClusterRoles'];
```